plugins-structure/packages/core/src/records/utils.js

```diff
@@ -117,8 +117,14 @@ export const validRecordStates = [
   ...validNumberRecordStates,
 ];
 
-export function getTokensToFetch({ records, pageSize, inventory }) {
-  let { lastTokenPos, tokens } = inventory;
+export function getTokensToFetch({
+  records,
+  pageSize,
+  inventory,
+  lastTokenPosName,
```

Comment:
What about `lastTokenPosKey` instead?
```suggestion
  lastTokenPosKey,
```